support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/SentinelEmbeddedTicketRegistryTests.java

```diff
@@ -0,0 +1,100 @@
+package org.apereo.cas.ticket.registry;
+
+import org.apereo.cas.config.CasCoreTicketCatalogConfiguration;
+import org.apereo.cas.config.CasCoreTicketsConfiguration;
+import org.apereo.cas.config.CasCoreWebConfiguration;
+import org.apereo.cas.config.RedisTicketRegistryConfiguration;
+import org.apereo.cas.config.support.CasWebApplicationServiceFactoryConfiguration;
+import org.apereo.cas.util.junit.DisabledIfContinuousIntegration;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.aop.AopAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.autoconfigure.RefreshAutoConfiguration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import redis.embedded.RedisSentinel;
+import redis.embedded.RedisServer;
+
+/**
+ * Unit test for {@link RedisTicketRegistry}.
+ *
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+@Tag("Redis")
+@SpringBootTest(classes = {
+    RedisTicketRegistryConfiguration.class,
+    RefreshAutoConfiguration.class,
+    CasCoreWebConfiguration.class,
+    AopAutoConfiguration.class,
+    CasCoreTicketsConfiguration.class,
+    CasCoreTicketCatalogConfiguration.class,
+    CasWebApplicationServiceFactoryConfiguration.class})
+@TestPropertySource(properties = {
+    "cas.ticket.registry.redis.host=localhost",
+    "cas.ticket.registry.redis.port=6320",
+    "cas.ticket.registry.redis.readFrom=MASTER",
+    "cas.ticket.registry.redis.pool.max-active=20",
+    "cas.ticket.registry.redis.sentinel.master=mymaster",
+    "cas.ticket.registry.redis.sentinel.node[0]=localhost:26739",
+    "cas.ticket.registry.redis.sentinel.node[1]=localhost:26740",
+    "cas.ticket.registry.redis.sentinel.node[2]=localhost:26741"
+})
+@EnableTransactionManagement(proxyTargetClass = true)
+@EnableAspectJAutoProxy(proxyTargetClass = true)
+@DisabledIfContinuousIntegration
+public class SentinelEmbeddedTicketRegistryTests extends BaseTicketRegistryTests {
```

Comment:
SentinelEmbeddedTicketRegistryTests seems identical to SentinelEmbeddedServiceRegistryTests. Perhaps duplications can be refactored to a common redis module and reused here?

Reply:
I'm shared to do that as It will create only a class that will permit to set redis+sentinel servers on ports that should be passed as we can't use the same ports. Also the two class doesn't extends the same classes. But I'm open for change, just let me know on which part you are thinking.

Reply:
I think you should be able to do something like this as an example:

```
@SpringBootTest(classes = {RedisServiceRegistryConfiguration.class, RefreshAutoConfiguration.class})
@EnableScheduling
@EnableConfigurationProperties(CasConfigurationProperties.class)
@Tag("Redis")
public abstract class BaseRedisSentinelServiceRegistryTests extends AbstractServiceRegistryTests {}
```

Then, you include the properties and `@DisabledIfContinuousIntegration` on the child class only to minimize duplication. If you examine ldap tests, some tend to do similar things.

If this looks too complicated, leave it be and we might refactor later as part of the general cleanup for tests by @tduehr.

Reply:
Ok I purposed a new commit with a refactored part.